aioxmpp/im/p2p.py

```diff
@@ -0,0 +1,179 @@
+        return self.__peer_jid
+
+    @property
+    def members(self):
+        return self.__members
+
+    @property
+    def me(self):
+        return self.__members[0]
+
+    @asyncio.coroutine
+    def send_message(self, msg):
+        msg.to = self.__peer_jid
+        self.on_message(msg, self.me, MessageSource.STREAM)
+        yield from self._client.stream.send(msg)
+
+    @asyncio.coroutine
+    def send_message_tracked(self, msg):
+        raise self._not_implemented_error("message tracking")
+
+    @asyncio.coroutine
+    def leave(self):
+        yield from super().leave()
+
+
+class Service(AbstractConversationService, aioxmpp.service.Service):
+    """
+    Manage one-to-one conversations.
+
+    This service manages one-to-one conversations, including private
+    conversations running in the framework of a multi-user chat. In those
+    cases, the respective multi-user chat conversation service requests a
+    conversation from this service to use.
+
+    For each bare JID, there can either be a single conversation for the bare
+    JID or zero or more conversations for full JIDs. Mixing conversations to
+    bare and full JIDs of the same bare JID is not allowed, because it is
+    ambiguous.
+
+    If bare JIDs are used, the conversation is assumed to be between
+
+    .. note::
+
+       This service does *not* automatically create new conversations when
+       messages which cannot be mapped to any conversation are incoming. This
+       is handled by the :class:`AutoOneToOneConversationService` service. The
+       reason for this is that it must have a lower priority than multi-user
+       chat services so that those are able to handle those messages if they
+       belong to a new private multi-user chat conversation.
+
+    """
+
+    ORDER_AFTER = [
+        ConversationService,
+        IMDispatcher,
+    ]
+
+    def __init__(self, client, **kwargs):
+        super().__init__(client, **kwargs)
+        self._conversationmap = {}
+        self.on_conversation_new.connect(
+            self.dependencies[ConversationService]._add_conversation
+        )
+
+    def _make_conversation(self, peer_jid):
+        result = Conversation(self, peer_jid, parent=None)
+        self._conversationmap[peer_jid] = result
+        print("_make_conversation", peer_jid)
+        self.on_conversation_new(result)
+        return result
+
+    @aioxmpp.service.depfilter(IMDispatcher, "message_filter")
+    def _filter_message(self, msg, peer, sent, source):
+        try:
+            existing = self._conversationmap[peer]
+        except KeyError:
+            try:
+                existing = self._conversationmap[peer.bare()]
+            except KeyError:
+                existing = None
+
+        if     ((msg.type_ == aioxmpp.MessageType.CHAT or
+                 msg.type_ == aioxmpp.MessageType.NORMAL) and
+                msg.body):
+            if existing is None:
+                existing = self._make_conversation(peer.bare())
```

Comment:
This makes a new conversation, contradicting the documentation, which says, that no new conversation is created automatically.